# HG changeset patch
# User Sean Young <seannz@gmail.com>
# Date 1310842025 18000
#      Sat Jul 16 13:47:05 2011 -0500
# Node ID 8dd68099308829d9ca9d86753484bbe67ae2d0ad
# Parent  1a3cdafcb2a5dea991aa94a4a5c986ecb448d474
codesprint: Wrote 6 tests for commutation_matrix.m

diff --git a/scripts/linear-algebra/commutation_matrix.m b/scripts/linear-algebra/commutation_matrix.m
--- a/scripts/linear-algebra/commutation_matrix.m
+++ b/scripts/linear-algebra/commutation_matrix.m
@@ -90,8 +90,30 @@ function k = commutation_matrix (m, n)
   k = zeros (m * n, m * n);
   for i = 1 : m
     for j = 1 : n
       k ((i - 1) * n + j, (j - 1) * m + i) = 1;
     endfor
   endfor
 
 endfunction
+
+%!test
+%! c = commutation_matrix(1,1);
+%! assert(c,1);
+
+%!test
+%! A = rand(3,5);
+%! vc = vec(A);
+%! vr = vec(A');
+%! c = commutation_matrix(3,5);
+%! assert(c*vc,vr);
+
+%!test
+%! A = rand(4,6);
+%! vc = vec(A);
+%! vr = vec(A');
+%! c = commutation_matrix(4,6);
+%! assert(c*vc,vr);
+
+%!error commutation_matrix(0,0);
+%!error commutation_matrix(1,0);
+%!error commutation_matrix(0,1);
